### DISCUSSION TITLE
RuntimeError: This event loop is already running

### DISCUSSION THREAD

**Discussion by dtrianab** [sec-edgar/sec-edgar#235] [Q&A]
Hello, I have installed the package as advised at #223, import seems to get the package, I have got a previous version installed which didn't work out for me so I run **pip install secedgar==0.4.0-alpha** and I can see it in pip list with v 0.4.0 alpha.

from secedgar.core import CompanyFilings, FilingType

10Q filings for Apple (ticker "aapl")
my_filings = CompanyFilings(cik_lookup="aapl", filing_type=FilingType.FILING_10Q, user_agent="Diego (user@mail.com)")
my_filings.save('APPL')

I ran into the error **RuntimeError: This event loop is already running**, 
Do you maybe know how can I overcome this? 

I use spyder 5.0.5 along anaconda with python  3.8.8, also tried in jupyter.

Thank you very much for your help

![imagen](https://user-images.githubusercontent.com/36299847/129351435-21157c70-ac7d-4be4-ae7e-e302858bd276.png)

**Comment by jackmoody11** [comment]
Hi @dtrianab I think this should be answered by #191. Running in Jupiter or iPython can lead to issues. My understanding is that Jupyter/iPython has an event loop running. You can either run it from a .py file or use nest_asyncio as explained [here](https://github.com/ipython/ipython/issues/11338). I am not sure if there is a way we are supposed to be checking if the event loop already exists, but am open to suggestions. Let me know if this helps you.

**Reply by dtrianab** [reply]
Hello @jackmoody11 thank you for youy quick reply, that actually helped :) . I updated kernel with ```pyhton pip install ipykernel ipython --upgrade ```  I also added the **nest_asyncio.apply()** which solved the event loop already running problem. Although, now I go into `
ClientResponseError: 403, message='Forbidden', url=URL('https://www.sec.gov/Archives/edgar/data/320193/000032019320000062/0000320193-20-000062.txt')` similarly to #207   it seems my request is refused, but it doesnt raise the exception for overhitting the sec as defined in client.py, it fails at last when invoking the 'check for status' in aiohttp, I have that packet on v 3.7.4.post0, it does not even process the first file, but the url is actually reachable in the browser. I also check the NetworkClient class and added smaller params to my Company Filings function ```CompanyFilings(cik_lookup="aapl",          filing_type=FilingType.FILING_10Q, user_agent="Diego (dtb_fk@hotmail.com)", rate_limit=1, batch_size=1)```, it also happens if I use cmd, my spyder is actually pointing to same python path compiler as cmd win10 (Python 3.8.8), so I doubt it has to do with environments, 
Do you maybe have a suggestion ?

```py
from secedgar.core import CompanyFilings, FilingType

import nest_asyncio
nest_asyncio.apply()

#10Q filings for Apple (ticker "aapl")
my_filings = CompanyFilings(cik_lookup="aapl", 
                            filing_type=FilingType.FILING_10Q, 
                            user_agent="Diego (dtb_fk@hotmail.com)"
                            )
my_filings.save('aapl')
```

```py
my_filings.save('aapl')
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x000001FE22C51820>
  0%|          | 0/280 [00:00<?, ?it/s]
Traceback (most recent call last):

  File "C:\Users\Diego\AppData\Local\Temp/ipykernel_19932/4077673180.py", line 1, in <module>
    my_filings.save('aapl')

  File "C:\ProgramData\Anaconda3\lib\site-packages\secedgar\core\company.py", line 244, in save
    loop.run_until_complete(self.client.wait_for_download_async(inputs))

  File "C:\ProgramData\Anaconda3\lib\site-packages\nest_asyncio.py", line 70, in run_until_complete
    return f.result()

  File "C:\ProgramData\Anaconda3\lib\asyncio\futures.py", line 178, in result
    raise self._exception

  File "C:\ProgramData\Anaconda3\lib\asyncio\tasks.py", line 280, in __step
    result = coro.send(None)

  File "C:\ProgramData\Anaconda3\lib\site-packages\secedgar\client.py", line 252, in wait_for_download_async
    await asyncio.gather(

  File "C:\ProgramData\Anaconda3\lib\asyncio\tasks.py", line 349, in __wakeup
    future.result()

  File "C:\ProgramData\Anaconda3\lib\asyncio\tasks.py", line 280, in __step
    result = coro.send(None)

  File "C:\ProgramData\Anaconda3\lib\site-packages\secedgar\client.py", line 221, in fetch_and_save
    contents = await self.fetch(link, session)

  File "C:\ProgramData\Anaconda3\lib\site-packages\secedgar\client.py", line 208, in fetch
    async with await session.get(link) as response:

  File "C:\ProgramData\Anaconda3\lib\site-packages\aiohttp\client.py", line 625, in _request
    resp.raise_for_status()

  File "C:\ProgramData\Anaconda3\lib\site-packages\aiohttp\client_reqrep.py", line 1000, in raise_for_status
    raise ClientResponseError(

ClientResponseError: 403, message='Forbidden', url=URL('https://www.sec.gov/Archives/edgar/data/320193/000032019320000010/0000320193-20-000010.txt')


```
